database/src/main/scala/cromwell/database/DbNullable.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.database
+
+/**
+  * Trait and typeclasses for converting to and from non-null representations of data in the database that is
+  * actually null in the domain model.
+  * This is used to work around https://bugs.mysql.com/bug.php?id=8173
+  */
+object DbNullable {
```

Comment:
This is now almost entirely vestigial from an earlier stage of the PR where there was a unique index on the (workflowId + callFqn + index + attempt + key + timestamp) columns.  I ended up removing this unique index due to complications in dealing with constraint violations plus the [MySQL ugliness](https://bugs.mysql.com/bug.php?id=8173) involved with nullable columns in unique constraints.  I'd like to tech talk this tomorrow if possible.